# Descriptions

This directory contains programs to detect HIV-1 integration sites (ISs) from raw fastq files generated by LM-PCR.
The analytical pipeline are described in detail elsewhere ([Satou et al, 2017](https://www.nature.com/articles/s41598-017-07307-4)).

## batch\_pipeline.sh
Main script file  
This script depends on the following environments and programs:
* Ubuntu (18.04.4 LTS)
* python2 (v2.7.17)
  * sys
  * subprocess
* cutadapt (v1.15)
* BWA (v0.7.17-r1188)
* bedtools (v2.27.0)
* samtools (v1.7)

### Description of output file (**test.cell_count.txt**):
1. Chrom: chromosome name
2. Position: genomic position
3. Strand: direction of HIV-1 IS
4. Count: the number of break points in each integration site. Theoretically, this number represents the clone size.


## programs
Directory containing programs used in **batch\_pipeline.sh**.


## test_data
Directory containing demo data (raw fastq file generated by LM-PCR).  
In addition, the final output (**test.cell_count.txt**) of this pipeline is also included.

